tests/unit/test_main.py

```diff
@@ -246,5 +248,69 @@ def test_preference_args(self):
         prefs = main.preference(None, prefs_args)
         self.assertEquals(len(prefs), 0)
 
+
+class TestCheckMozregresionVersion(unittest.TestCase):
+    @patch('requests.get')
+    @patch('sys.stdout')
+    def test_pypi_is_accesible(self, stdout, get):
+        options = None
+        logger = Mock('mozlog.structured.commandline.setup_logging')(
+                      "mozregression",
+                      options,
+                      {"mach": stdout})
+        response = Mock(json=lambda: {'info': {'version':  __version__}})
+        get.return_value = response
+        main.check_mozregression_version(logger)
+        self.assertFalse(logger.critical.called)
+
+    @patch('requests.get')
+    @patch('sys.stdout')
+    def test_RequestException_error(self, stdout, get):
+        options = None
+        logger = Mock('mozlog.structured.commandline.setup_logging')(
+                      "mozregression",
+                      options,
+                      {"mach": stdout})
+        get.side_effect = requests.RequestException
+        main.check_mozregression_version(logger)
+        self.assertRaises(requests.RequestException, get)
+
+    @patch('requests.get')
+    @patch('sys.stdout')
+    def test_key_error(self, stdout, get):
+        options = None
+        logger = Mock('mozlog.structured.commandline.setup_logging')(
+                      "mozregression",
+                      options,
+                      {"mach": stdout})
+        get.side_effect = KeyError
+        main.check_mozregression_version(logger)
+        self.assertRaises(KeyError, get)
+
+    @patch('requests.get')
+    @patch('sys.stdout')
+    def test_value_error(self, stdout, get):
+        options = None
+        logger = Mock('mozlog.structured.commandline.setup_logging')(
+                      "mozregression",
+                      options,
+                      {"mach": stdout})
+        get.side_effect = ValueError
+        main.check_mozregression_version(logger)
+        self.assertRaises(ValueError, get)
+
+    @patch('requests.get')
+    @patch('sys.stdout')
+    def test_warning_is_called_twice(self, stdout, get):
```

Comment:
I don't really like the name here. Maybe "test_warn_if_version_not_up_to_date" ?